packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -134,7 +135,55 @@ class BasketSpreadPriceFeed extends PriceFeedInterface {
       return this._getSpreadFromBasketPrices(experimentalPrices, baselinePrices, denominatorPrice);
     }
   }
+  // This searches for closest time in a list of [time,price] data. Based on code in affiliates models/prices.
+  closestTime(list) {
+    return time => {
+      const result = list.reduce((a, b) => {
+        const aDiff = Math.abs(a[0] - time);
+        const bDiff = Math.abs(b[0] - time);
+
+        // if differences are equal, return larger? timestamp
+        if (aDiff == bDiff) {
+          return a < b ? a : b;
+        }
+        // if diffs are diff, return smallest diff
+        return bDiff < aDiff ? b : a;
+      });
+      assert(result, "no closest time found");
+      return this.toBN(result[1]);
+    };
+  }
+  // This function does something similar to get historicalprice, but does not have the luxury of only caring about a
+  // single point in time. It has to run the basketspread price across all timestamps available. This is complicated
+  // as there are multiple price histories which we must search through at each matching timestamp to find the closets
+  // prices to add into the basket calculation.
+  // Returns data in the form of [[time,price]]
+  getHistoricalPricePeriods() {
```

Comment:
this function was refactored and its quite complicated. the idea here is that there are multiple historical price feeds. its possible these feeds all have different timestamps as well. we need to somehow go through the time, pull the nearest price from each feed, then pass it into the `getSpreadFromBasketPrices` function. 

In order to do this, ive added this `closestTime` function as a  helper to search for closest timestamp within each price feed.